go/tools/builders/nogo_typeparams_go117.go

```diff
@@ -0,0 +1,23 @@
+/* Copyright 2022 The Bazel Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//go:build !go1.18
+// +build !go1.18
```

Comment:
I think we have been shipping some features in rules_go that are exclusively to 1.18 and are not backward compatible. 🤔 

I don't know if we need to have this file at all 🤔  It would massively simplify this PR down to a single line code change.

Reply:
From the discussion on #3231, I gather that we want to keep the build-tagged files for 1.17 and 1.18, at the very least until 1.19 is released, correct?

Reply:
Yes. We need to stay compatible with Go 1.17 at least until it's deprecated, normally after 1.19 is released.